x-pack/platform/plugins/shared/ml/public/application/model_management/delete_models_modal.tsx

```diff
@@ -61,33 +63,27 @@ export const DeleteModelsModal: FC<DeleteModelsModalProps> = ({ models, onClose
   }, 0);
 
   const deleteModels = useCallback(async () => {
-    try {
-      await Promise.all(
-        modelIds.map((modelId) =>
-          trainedModelsApiService.deleteTrainedModel(modelId, {
-            with_pipelines: deletePipelines,
-            force: pipelinesCount > 0,
-          })
-        )
-      );
-    } catch (error) {
-      displayErrorToast(
-        error,
-        i18n.translate('xpack.ml.trainedModels.modelsList.fetchDeletionErrorMessage', {
-          defaultMessage: '{modelsCount, plural, one {Model} other {Models}} deletion failed',
-          values: {
-            modelsCount: modelIds.length,
-          },
-        })
-      );
-    }
-    onClose(true);
+    setIsDeleting(true);
+    trainedModelsService
+      .deleteModels(modelIds, {
+        with_pipelines: deletePipelines,
+        force: pipelinesCount > 0,
+      })
+      .subscribe({
+        next: () => {
+          onClose(false, true);
+        },
+        error: () => {
+          setIsDeleting(false);
+        },
+      });
```

Comment:
making `deleteModels` should simplify the logic code here as well 
```suggestion
    setIsDeleting(true);
    try {
      await trainedModelsService.deleteModels(modelIds, {
        with_pipelines: deletePipelines,
        force: pipelinesCount > 0,
      });
      onClose(false, true);
    } catch () {
       setIsDeleting(false);
    }
```